### PR TITLE
[CN-262] Add Kubernetes provider info into platform struct

### PR DESCRIPTION
**Pull request by mtyazici**
Changes:
- Added provider info to Platform struct, providers are `EKS`, `GKE` and `AKS`.
- Added config to the Platform resolver client. It used to use Kubeconfig as default one, I set its config to the Manager's one.

### CODE REVIEW

File: controllers/platform/platform.go [hunk 1]
```diff
@@ -5,17 +5,26 @@ import (
...
 )
 
-type Platform struct {
-	Type    PlatformType `json:"type"`
-	Version string       `json:"version"`
-}
+type ProviderType string
+
+const (
+	AWS ProviderType = "AWS"
```
Comment (hasancelik) [inline controllers/platform/platform.go:24]: It should be `EKS`

**Review by hasancelik — CHANGES_REQUESTED** [review]

File: controllers/platform/platform.go (hunk 1 above)
Comment (mtyazici) [reply controllers/platform/platform.go]: Fixed it, thanks

File: controllers/platform/platform.go [hunk 2]
```diff
@@ -29,11 +44,24 @@ func GetPlatform() (Platform, error) {
 	var err error
 	plt, err = getInfo()
 	if err != nil {
-		return plt, err
```
Comment (hakanmemisoglu) [inline controllers/platform/platform.go:32]: I'd convert this to return `nil` and change the function signature.

**Review by hakanmemisoglu — APPROVED** [review]
LGTM. Just a small comment...

File: controllers/platform/platform.go (hunk 2 above)
Comment (mtyazici) [reply controllers/platform/platform.go:32]: I agree with returning nil but what do you mean by changing function signature? Do you mean returning pointers?

**Comment by mtyazici** [comment]
@hakanmemisoglu and @hasancelik,

I am thinking of setting the platform struct only at the operator start up. The struct will be set and consequent `get`s only will return the field. However, this requires that if the platform information cannot be set at startup the operator will exit.

The only reason I did it this way is because the Kubernetes version could get updated and operator would not be aware of that. However, operator will be restarted when there is a Kubernetes version update. So getting the Platform information only at startup is enough. WDYT?

**Comment by hasancelik** [comment]
@mtyazici 

Good approach 👍  The operator has to retrieve Platform(Kubernetes | OCP) information during start-up because we modify our resources according to this info. So exit scenario should not be a problem for us.

**Review by hasancelik — APPROVED** [review]

**Pull request merged**